packages/plugin-github-pr-link/src/commands/link-github-pr.ts

```diff
@@ -0,0 +1,62 @@
+import { Command, Flags, Interfaces } from '@oclif/core'
+import { BaseCommand, composeFlags, envIdFlags } from '@preevy/cli-common'
+import { Octokit } from 'octokit'
+import { FlatTunnel, findAmbientEnvId } from '@preevy/core'
+import { ParsedFlags, flagsDef } from '../flags'
+import { PluginConfig, loadGithubConfig } from '../config'
+import { upsertPreevyComment } from '../lib/github-comment'
+
+// eslint-disable-next-line no-use-before-define
+export type Flags<T extends typeof Command> = Interfaces.InferredFlags<typeof LinkGithubPr['baseFlags'] & T['flags']>
+export type Args<T extends typeof Command> = Interfaces.InferredArgs<T['args']>
+
+// eslint-disable-next-line no-use-before-define
+class LinkGithubPr extends BaseCommand<typeof LinkGithubPr> {
+  static id = 'link-github-pr'
+  static description = 'Link a GitHub Pull Request to an existing environment'
+
+  static flags = {
+    ...composeFlags,
+    ...envIdFlags,
+    ...flagsDef,
+  }
+
+  protected get pluginConfig(): PluginConfig {
+    return (this.constructor as unknown as { pluginConfig: PluginConfig }).pluginConfig
+  }
+
+  async run() {
+    const config = await loadGithubConfig(this.pluginConfig, this.flags as ParsedFlags)
+
+    if (!config) {
+      throw new Error('No GitHub config provided for plugin - specify env vars, flags or populate the compose file')
+    }
+
+    const log = this.logger
+    const projectName = (await this.ensureUserModel()).name
+    log.debug(`project: ${projectName}`)
+    const envId = this.flags.id || await findAmbientEnvId(projectName)
+    log.debug(`envId: ${envId}`)
+
+    const urls = await this.config.runCommand('urls', [
```

Comment:
I'm a bit confused.
This plugin can be used both as a hook and as a command?
In the command case, the logic is a bit weird as it uses the urls, couldn't it enable the plugin hooks for this run.

Reply:
Plugins can add commands and flags.
This command is separate from `up` and doesn't include its logic - it just upserts the GitHub comment.

Reply:
So, it's to set an example?

Reply:
Or its worth keeping for testing?

Reply:
I worry that it might cause confusion